compiler/backend/presLangScript.sml

```diff
@@ -131,8 +135,63 @@ val mod_to_pres_prompt_def = Define`
 val mod_to_pres_def = Define`
   mod_to_pres prompts = Prog (MAP mod_to_pres_prompt prompts)`;
 
+(* con_to_pres *)
+val con_to_pres_pat_def = tDefine"con_to_pres_pat"`
+  con_to_pres_pat p =
+    case p of
+       | conLang$Pvar varN => presLang$Pvar varN
+       | Plit lit => Plit lit
+       | Pcon opt ps => ConPcon opt (MAP con_to_pres_pat ps)
+       | Pref pat => Pref (con_to_pres_pat pat)`
+    cheat;
+
+val con_to_pres_exp_def = tDefine"con_to_pres_exp"`
+  (con_to_pres_exp (conLang$Raise t e) = Raise t (con_to_pres_exp e))
+  /\ 
+  (con_to_pres_exp (Handle t e pes) = Handle t (con_to_pres_exp e) (con_to_pres_pes pes))
+  /\
+  (con_to_pres_exp (Lit t l) = Lit t l)
+  /\
+  (con_to_pres_exp (Con t ntOpt exps) = ConCon t ntOpt (MAP con_to_pres_exp
+  exps))
+  /\ 
+  (con_to_pres_exp (Var_local t varN) = Var_local t varN)
+  /\
+  (con_to_pres_exp (Var_global t num) = Var_global t num)
+  /\
+  (con_to_pres_exp (Fun t varN e) = Fun t varN (con_to_pres_exp e))
+  /\
+  (con_to_pres_exp (App t op exps) = App t (Conlang op) (MAP con_to_pres_exp exps))
+  /\
+  (con_to_pres_exp (Mat t e pes) = Mat t (con_to_pres_exp e) (con_to_pres_pes pes))
+  /\
+  (con_to_pres_exp (Let t varN e1 e2) = Let t varN (con_to_pres_exp e1)
+  (con_to_pres_exp e2))
+  /\ 
+  (con_to_pres_exp (Letrec t funs e) = Letrec t (MAP (\(v1,v2,e).(v1,v2,con_to_pres_exp e)) funs) (con_to_pres_exp e))
+  /\
+  (con_to_pres_exp (Extend_global t num) = Extend_global t num)
+  /\ 
+  (con_to_pres_pes [] = [])
+  /\
+  (con_to_pres_pes ((p,e)::pes) =
+    (con_to_pres_pat p, con_to_pres_exp e)::con_to_pres_pes pes)`
+  cheat; 
+
+val con_to_pres_dec_def = Define`
+  con_to_pres_dec d =
+    case d of
+       | conLang$Dlet num exp => presLang$Dlet num (con_to_pres_exp exp)
+       | Dletrec funs => Dletrec (MAP (\(v1,v2,e). (v1,v2,con_to_pres_exp e)) funs)`; 
+
+val con_to_pres_prompt_def = Define`
+  con_to_pres_prompt (Prompt decs) = Prompt NONE (MAP con_to_pres_dec decs)`;
+
+val con_to_pres_def = Define`
+  con_to_pres prompts = Prog (MAP con_to_pres_prompt prompts)`;
+
 (* pres_to_json *)
-(* TODO: Add words *)
+(* TODO: Add words, add ConCon and ConPcon *)
```

Comment:
Isn't all these necessary for this to work? Can we really be without it, even for now?